src/components/structures/RoomView.js

```diff
@@ -1412,6 +1260,33 @@ module.exports = React.createClass({
                     </div>
             }
 
+
+            // if we have search results, we keep the messagepanel (so that it preserves its
+            // scroll state), but hide it.
+            var searchResultsPanel;
+            var hideMessagePanel = false;
+
+            if (this.state.searchResults) {
+                searchResultsPanel = (
+                    <ScrollPanel ref="searchResultsPanel" className="mx_RoomView_messagePanel"
+                            onFillRequest={ this.onSearchResultsFillRequest }>
+                        <li className={scrollheader_classes}></li>
+                        {this.getSearchResultTiles()}
+                    </ScrollPanel>
+                );
+                hideMessagePanel = true;
+            }
+
+            var messagePanel = (
```

Comment:
Keeping it hidden is fine IMO, particularly for something as gnarly as scroll positions. If it becomes a performance bottleneck *then* we can re-visit this.